### PR TITLE
Pin refactor and RAMPS support

### PR DESCRIPTION
**Pull request by michielbaird**
Hey I changed the pins structure to make it easier to maintain other forms of hardware. And added RAMPS as a proof of concept. 

Let me know if I need to fix something to make it mergable.

### CODE REVIEW

File: Firmware/util.cpp [hunk 1]
```diff
@@ -1,288 +1,288 @@
...
-    uint16_t ver_current[4];
-    if (parse_version_P(FW_VERSION_STR, ver_current)) {
-        eeprom_update_word((uint16_t*)EEPROM_FIRMWARE_VERSION_MAJOR,    ver_current[0]);
-        eeprom_update_word((uint16_t*)EEPROM_FIRMWARE_VERSION_MINOR,    ver_current[1]);
-        eeprom_update_word((uint16_t*)EEPROM_FIRMWARE_VERSION_REVISION, ver_current[2]);
-        // See FirmwareRevisionFlavorType for the definition of firmware flavors.
-        eeprom_update_word((uint16_t*)EEPROM_FIRMWARE_VERSION_FLAVOR,   ver_current[3]);
-    }
-}
+#include "Configuration.h"
```
Comment (michielbaird) [inline Firmware/util.cpp]: I didnt actually change anything here beyond BEEPER -> BEEPER_PIN must be a newline issue

**Comment by GurliGebis** [comment]
You have changed temperature.cpp from DOS line endings to UNIX line endings, resulting in every single line of the file being changed.

Could you change it back, and add a commit for that?

You can do it using the unix2dos utility, or most editors will allow you to do when saving.

File: Firmware/util.cpp (hunk 1 above)
Comment (GurliGebis) [reply Firmware/util.cpp]: Indeed, you can convert it the unix2dos command - the same goes for temperature.cpp

**Comment by GurliGebis** [comment]
Let me know if you need help with it - I'll fix it on my repo and make a pull request for you to pull to get the fix into your repo.

**Comment by michielbaird** [comment]
Much better! Sorry for that screw up :) If you do decide to merge it use the squash option to avoid it overriding the history.

**Review by GurliGebis — APPROVED** [review]
Great idea, and it looks clean to me.

**Comment by GurliGebis** [comment]
I haven't got write access to the repo, so I cannot merge it. But from what I can see, it should be a clean merge, without breaking anything.

**Comment by bvarner** [comment]
I see this PR has been floating around for quite some time. I'm _very_ interested in either this (being able to run the Prusa firmware on RAMPS) or back-porting more of the Prusa firmware changes into Marlin.

Either way, how can I help?

**Comment by cimoalpacino** [comment]
> 
> 
> I see this PR has been floating around for quite some time. I'm _very_ interested in either this (being able to run the Prusa firmware on RAMPS) or back-porting more of the Prusa firmware changes into Marlin.
> 
> Either way, how can I help?

if you like, there's the option
https://github.com/cimoalpacino/Prusa-Firmware/tree/MK3_3.8.0_RAMPS

**Comment by 3d-gussner** [comment]
Closed as MK2 branch isn't maintained anymore. Thanks

**Pull request closed without merging**